### PR TITLE
Parsley instance selector to master

### LINKED ISSUE #137: Add Parsley Instance Selector
Sometimes, we want to send messages to a specific instance of parsley, rather than all of them at once. Adding this feature to the dashboard in a nice way would be nice.

There are currently 2, rather bad ways to do this on the `wdr` branch. We'd like a fusion of the two (taking the non-bad parts from both)

### PR DESCRIPTION
**Pull request by MaisimZaman**
## Description
Implemented parsley instance selector in the Menu Bar under "CAN" that allows the user to select which particular instance of parsley the CAN sender will uniquely send data to  

This PR closes #137 

## How to Test
-Start omnibus with: `python -m omnibus`

-If testing with personal computer /laptop ports comment theses lines out in `sources/parsley/main.py` in SerialCommunicator: 

```
class SerialCommunicator:
    def __init__(self, port, baud, timeout):
        #self.port = port
        #self.serial = serial.Serial(port, baud, timeout=timeout)
        pass
    def read(self):
        return b''
        return self.serial.read(4096)

    def write(self, msg):
        #self.serial.write(msg)
        pass
```

-Now run multiple sinks of parsley by running `python sources/parsley/main.py COM3`, COM3 can be any number (3. 4, 5. 6). Remember to run each sink on an individual terminal

-Finally start the dashboard with: `python sinks/dashboard/main.py` to test the changes in this PR

- - -
This change is [Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181)

### CODE REVIEW

**Review by Darth-Raazi — COMMENTED** [review]
@MaisimZaman please fill out the PR template so that others can know what your changes are and how to test them

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-:-NkqmjUGBGeDJHl046dP:bbvz7nz)* status: 0 of 5 files reviewed, all discussions resolved (waiting on @BluCodeGH and @KavinSatheeskumar)

File: sinks/dashboard/dashboard.py [hunk 1]
```diff
@@ -218,7 +235,61 @@ def return_fun():
...
+   
+       
+        our_lst = [e[15:]
+                    for e in publisher.get_all_streams() if e.startswith("Parsley health ")]
+        
+        
+        if self.current_parsley_instances != our_lst or self.refresh_track % 2 == 0:
+            self.can_selector.clear()
+            
+            if (self.refresh_track % 2 == 0):
```
[the remark this answers is not in the document]
Comment (MaisimZaman) [reply sinks/dashboard/dashboard.py]: Okay, I changed it to a Boolean and pushed the changes

**Review by Darth-Raazi — CHANGES_REQUESTED** [review]
Great work!

> Reviewed 3 of 5 files at r1, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-:-NmLhpn_8yKdFvQ9j5X7:b-9dwnzc)* status: all files reviewed, 12 unresolved discussions (waiting on @BluCodeGH, @KavinSatheeskumar, and @MaisimZaman)

___
*[`sinks/dashboard/dashboard.py` line 91 at r4](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-NmLf34vE8e--dQ_x2Qu:-NmLf34vE8e--dQ_x2Qv:b-ti7mlt) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/2ab0f4f8d933c255c409195122015a78a35d1abe/sinks/dashboard/dashboard.py#L91)):*
> ```Python
> 
>         self.current_parsley_instances = []
>         
> ```

This blank line can be deleted
___
*[`sinks/dashboard/dashboard.py` line 162 at r4](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-NmLgmz19GK3fce67vLW:-NmLgmz19GK3fce67vLX:bug81m5) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/2ab0f4f8d933c255c409195122015a78a35d1abe/sinks/dashboard/dashboard.py#L162)):*
> ```Python
> 
>         # adding a button to switch instances of parsley
>         self.can_selector = menubar.addMenu("CAN")
> ```

I think it would be better to name the menu `Parsley` rather than `CAN` to make it more clear for the user
___
*[`sinks/dashboard/dashboard.py` line 244 at r4](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-NmLfDc8ClfSJHKnr0QT:-NmLfDc8ClfSJHKnr0QU:b-fzuc69) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/2ab0f4f8d933c255c409195122015a78a35d1abe/sinks/dashboard/dashboard.py#L244)):*
> ```Python
>     def set_parsley_to_none(self):
>         self.parsley_instance = 'None'
>         self.refresh_track = True
> ```

No need for a second function for none, see below
___
*[`sinks/dashboard/dashboard.py` line 255 at r4](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-NmLfID-9pJKUs3E74Ms:-NmLfID-9pJKUs3E74Mt:bx4cw1y) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/2ab0f4f8d933c255c409195122015a78a35d1abe/sinks/dashboard/dashboard.py#L255)):*
> ```Python
>        
>         our_lst = [e[15:]
>                     for e in publisher.get_all_streams() if e.startswith("Parsley health ")]
> ```

It would be better to name this list something like `parsley_streams` rather than a generic `our_lst`. Also you don't need the extra space after `Parsley health` and it can be one line
___
*[`sinks/dashboard/dashboard.py` line 256 at r4](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-NmLgLTmEX6U9igfbBjf:-NmLgLTmEX6U9igfbBjg:b-bhgcqy) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/2ab0f4f8d933c255c409195122015a78a35d1abe/sinks/dashboard/dashboard.py#L256)):*
> ```Python
>         our_lst = [e[15:]
>                     for e in publisher.get_all_streams() if e.startswith("Parsley health ")]
>         
> ```

Append `"None"` to `our_lst` here. See above and below comments
___
*[`sinks/dashboard/dashboard.py` line 271 at r4](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-NmLfnl_3iO_URpwXNM5:-NmLfnl_3iO_URpwXNM6:bsokfef) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/2ab0f4f8d933c255c409195122015a78a35d1abe/sinks/dashboard/dashboard.py#L271)):*
> ```Python
>                 new_action.setCheckable(True)
>                 
>                 if self.parsley_instance != our_lst[inst]:
> ```

This is a nitpick, but I prefer having the success or true condition in the if statement rather than the fail/false condition. Basically instead of `!= ... setChecked(False) else ...` do `== ... setChecked(True) else ...`
___
*[`sinks/dashboard/dashboard.py` line 285 at r4](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-NmLgbI346-SwMTimGaE:-NmLgbI346-SwMTimGaF:bjf3jwl) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/2ab0f4f8d933c255c409195122015a78a35d1abe/sinks/dashboard/dashboard.py#L285)):*
> ```Python
>                     else:
>                         none_action.setChecked(False)
>                     self.lockableActions.append(none_action)
> ```

This entire section can be removed if `"None"` is already in the list. See above comments
___
*[`sinks/dashboard/items/can_sender.py` line 20 at r4](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-NmLh5o75QXxJNbsmKXJ:-NmLh5o75QXxJNbsmKXK:b-195h9n) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/2ab0f4f8d933c255c409195122015a78a35d1abe/sinks/dashboard/items/can_sender.py#L20)):*
> ```Python
> from .dashboard_item import DashboardItem
> from utils import EventTracker
> ```

No blank line needed here
___
*[`sinks/dashboard/items/periodic_can_sender.py` line 7 at r4](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-NmLhOMpAsPU1tT_JEcl:-NmLhOMpAsPU1tT_JEcm:b-195h9n) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/2ab0f4f8d933c255c409195122015a78a35d1abe/sinks/dashboard/items/periodic_can_sender.py#L7)):*
> ```Python
> from .dashboard_item import DashboardItem
> from .registry import Register
> ```

No blank line needed here
___
*[`sources/parsley/main.py` line 24 at r4](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-NmLhauHCx18RZNf8Eih:-NmLhauHCx18RZNf8Eii:bbv0mlm) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/2ab0f4f8d933c255c409195122015a78a35d1abe/sources/parsley/main.py#L24)):*
> ```Python
>         self.serial = serial.Serial(port, baud, timeout=timeout)
> 
>         
> ```

Keep one blank line between methods
___
*[`sources/parsley/main.py` line 33 at r4](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-NmLhgAP1dAKNT91u4fP:-NmLhgAP1dAKNT91u4fQ:b2tw175) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/2ab0f4f8d933c255c409195122015a78a35d1abe/sources/parsley/main.py#L33)):*
> ```Python
>        
> 
>         
> ```

Only 2 blank lines needed here

**Review by Darth-Raazi — COMMENTED** [review]
Your PR's auto-test job is failing because of autopep8. Before committing and pushing, please run `./tools/format.sh`

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-:-NmLiHWdD4gOwgPLEIyx:bafneqn)* status: all files reviewed, 12 unresolved discussions (waiting on @BluCodeGH, @KavinSatheeskumar, and @MaisimZaman)

**Review by Darth-Raazi — CHANGES_REQUESTED** [review]
Still failing, please run this. Other than that, code looks good and it works!

> Reviewed 5 of 5 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-:-NmxntKj3bF3o3_sZ7oA:b-damazn)* status: all files reviewed, 3 unresolved discussions (waiting on @BluCodeGH, @KavinSatheeskumar, and @MaisimZaman)

___
*[`sinks/dashboard/dashboard.py` line 255 at r4](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-NmLfID-9pJKUs3E74Ms:-NmxolUp7SAUSuqbssAM:bzabrrt) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/2ab0f4f8d933c255c409195122015a78a35d1abe/sinks/dashboard/dashboard.py#L255)):*
<details><summary><i>Previously, Darth-Raazi (Ozayr Raazi) wrote…</i></summary><blockquote>

It would be better to name this list something like `parsley_streams` rather than a generic `our_lst`. Also you don't need the extra space after `Parsley health` and it can be one line
</blockquote></details>

Please address the second part of this comment

**Review by Darth-Raazi — APPROVED** [review]
:lgtm:, great work

> Reviewed 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/181#-:-NnBRVM78OrgWQArx5JU:bkxhlzd)* status: all files reviewed, 1 unresolved discussion (waiting on @BluCodeGH and @KavinSatheeskumar)

**Pull request merged**